src/XML/src/FMI3/fmi3_xml_model_structure.c

```diff
@@ -477,103 +489,96 @@ int fmi3_xml_handle_ContinuousStateDerivative(fmi3_xml_parser_context_t *context
         }
     }
     return 0;
 }
 
-int fmi3_xml_handle_ClockedState(fmi3_xml_parser_context_t *context, const char* data) {
+int fmi3_xml_handle_ClockedState(fmi3_xml_parser_context_t* context, const char* data) {
+    fmi3_xml_model_description_t* md = context->modelDescription;
+    fmi3_xml_model_structure_t* ms = md->modelStructure;
     if (!data) {
-        fmi3_xml_model_description_t* md = context->modelDescription;
-        fmi3_xml_model_structure_t* ms = md->modelStructure;
-
-        /* perform the parsing */
         if (fmi3_xml_parse_unknown(context, fmi3_xml_elmID_ClockedState,
                                    &ms->clockedStates, ms->clockedStateDeps)) {
             return -1;
         }
-
-        /* validate return values */
+    } else {
+        // post-processing
         fmi3_xml_variable_t* clockVar = (fmi3_xml_variable_t*)jm_vector_get_last(jm_voidp)(&ms->clockedStates);
 
         // previous attribute is required, spec: "If present, this variable is a <ClockedState>" */
         if (!clockVar->hasPrevious) {
-            ms->isValidFlag = 0;
-            fmi3_xml_parse_error(context,
+            fmi3_xml_parse_warning(context,
                     "The variable '%s' is a ClockedState, but does not define the attribute 'previous'.",
                     fmi3_xml_get_variable_name(clockVar));
-            return -1;
         }
 
         // ClockedStates must also have variability='discrete', and the attribute 'clocks'.
         // The same applies for variables with the attribute 'previous'.
-        // ClockedStates require the attribute 'previous' , so this is already checked.
+        // ClockedStates require the attribute 'previous', so this is already checked.
 
         /* must not be of base type fmi3Clock */
         if (fmi3_xml_get_variable_base_type(clockVar) == fmi3_base_type_clock) {
-            ms->isValidFlag = 0;
-            fmi3_xml_parse_error(context,
+            fmi3_xml_parse_warning(context,
                     "The variable '%s' is a ClockedState, but has the base type 'fmi3Clock'.",
                     fmi3_xml_get_variable_name(clockVar));
-            return -1;
         }
     }
     return 0;
 }
 
-int fmi3_xml_handle_InitialUnknown(fmi3_xml_parser_context_t *context, const char* data) {
+int fmi3_xml_handle_InitialUnknown(fmi3_xml_parser_context_t* context, const char* data) {
+    fmi3_xml_model_description_t* md = context->modelDescription;
+    fmi3_xml_model_structure_t* ms = md->modelStructure;
     if (!data) {
-        fmi3_xml_model_description_t* md = context->modelDescription;
-        fmi3_xml_model_structure_t* ms = md->modelStructure;
-
-        return fmi3_xml_parse_unknown(context, fmi3_xml_elmID_InitialUnknown, &ms->initialUnknowns, ms->initialUnknownDeps);
+        if (fmi3_xml_parse_unknown(context, fmi3_xml_elmID_InitialUnknown, &ms->initialUnknowns, ms->initialUnknownDeps)) {
+            return -1;
+        }
+    } else {
+        // post-processing
```

Comment:
I guess no post-processing is necessary is that correct?

Reply:
Correct. Respectively, none are implemented yet. 

Just added it as a placeholder here, for consistency will all the other ModelStructure elements.